```
Did not find CMake 'cmake'
Found CMake: NO
Run-time dependency cjson found: NO (tried pkgconfig and cmake)
Has header "cjson/cJSON.h" : NO
Message: Using builtin cJSON library
Message: Building mbedtls
Run-time dependency mbedcrypto found: NO (tried pkgconfig and cmake)
Has header "mbedtls/aes.h" : NO
Message: Using builtin mbedtls code
Configuring version.h using configuration
Program objcopy found: YES
Configuring config.h using configuration
Message: Building tools
Library cmocka found: NO
test/rist/meson.build:72: DEPRECATION: ":" is not allowed in test name "Main profile encryption receive server mode unencrupted, sender: client mode", it has been replaced with "_"
test/rist/meson.build:75: DEPRECATION: ":" is not allowed in test name "Main profile encryption client mode, sender: server mode unencrypted", it has been replaced with "_"
Build targets in project: 7
NOTICE: Future-deprecated features used:
 * 0.56.0: {'meson.source_root'}
```

Currently builds successful without cmake, if cmake enabled this error occurred which I couldn't solve/find it's root:

```
this derivation will be built:
  /nix/store/9zppx79cf2xzw0ff5bv9ri1qg2dm5g8n-librist-0.2.6.drv
this path will be fetched (9.89 MiB download, 41.34 MiB unpacked):
  /nix/store/aw946f68hy6y76q06aw9mw0fqjrzwk4q-cmake-3.24.1
copying path '/nix/store/aw946f68hy6y76q06aw9mw0fqjrzwk4q-cmake-3.24.1' from 'https://cache.nixos.org'...
building '/nix/store/9zppx79cf2xzw0ff5bv9ri1qg2dm5g8n-librist-0.2.6.drv'...
unpacking sources
unpacking source archive /nix/store/i4y48pa96akwiaikcj2alhh5ywww1n52-source
source root is source
patching sources
configuring
fixing cmake files...
cmake flags: -GNinja -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/z9km10xa42m31b0yznjdd10mrhygvi88-librist-0.2.6/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/z9km10xa42m31b0yznjdd10mrhygvi88-librist-0.2.6/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/z9km10xa42m31b0yznjdd10mrhygvi88-librist-0.2.6/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/z9km10xa42m31b0yznjdd10mrhygvi88-librist-0.2.6/share/doc/librist -DCMAKE_INSTALL_INFODIR=/nix/store/z9km10xa42m31b0yznjdd10mrhygvi88-librist-0.2.6/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/z9km10xa42m31b0yznjdd10mrhygvi88-librist-0.2.6/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/z9km10xa42m31b0yznjdd10mrhygvi88-librist-0.2.6/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/z9km10xa42m31b0yznjdd10mrhygvi88-librist-0.2.6/include -DCMAKE_INSTALL_SBINDIR=/nix/store/z9km10xa42m31b0yznjdd10mrhygvi88-librist-0.2.6/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/z9km10xa42m31b0yznjdd10mrhygvi88-librist-0.2.6/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/z9km10xa42m31b0yznjdd10mrhygvi88-librist-0.2.6/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_SYSROOT= -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/j061mvdxw70xz20r2r0lr5lk0rcsjf9i-gcc-wrapper-11.3.0/bin/strip -DCMAKE_RANLIB=/nix/store/j061mvdxw70xz20r2r0lr5lk0rcsjf9i-gcc-wrapper-11.3.0/bin/ranlib -DCMAKE_AR=/nix/store/j061mvdxw70xz20r2r0lr5lk0rcsjf9i-gcc-wrapper-11.3.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/z9km10xa42m31b0yznjdd10mrhygvi88-librist-0.2.6
CMake Warning:
  Ignoring extra path from command line:

   ".."


CMake Error: The source directory "/build/source" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.

``

--- a/pkgs/development/libraries/librist/default.nix
+++ b/pkgs/development/libraries/librist/default.nix
@@ -1,0 +1,43 @@
+{ stdenv
+, lib
+, fetchFromGitLab
+, meson
+, ninja
+, pkg-config
+, cjson
+, cmocka
+, mbedtls
+}:
+
+stdenv.mkDerivation rec {
+  pname = "librist";
+  version = "0.2.7";
+
+  src = fetchFromGitLab {
+    domain = "code.videolan.org";
+    owner = "rist";
+    repo = "librist";
+    rev = "v${version}";
+    sha256 = "sha256-qQG2eRAPAQgxghMeUZk3nwyacX6jDl33F8BWW63nM3c=";
+  };
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+  ];
+
+  buildInputs = [
+    cjson
+    cmocka
+    mbedtls
+  ];
+
+  meta = with lib; {
+    description = "A library that can be used to easily add the RIST protocol to your application.";
+    homepage = "https://code.videolan.org/rist/librist";
+    license = with licenses; [ bsd2 mit isc ];
+    maintainers = with maintainers; [ raphaelr sebtm ];
+    platforms = platforms.all;
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -25981,6 +25981,8 @@ with pkgs;
 
   libratbag = callPackage ../os-specific/linux/libratbag { };
 
+  librist = callPackage ../development/libraries/librist { };
+
   libre-baskerville = callPackage ../data/fonts/libre-baskerville { };
 
   libre-bodoni = callPackage ../data/fonts/libre-bodoni { };

